test/parallel/test-require-node-prefix.js

```diff
@@ -0,0 +1,39 @@
+'use strict';
+
+require('../common');
+const assert = require('assert');
+const fs = require('fs');
+
+// For direct use of require expressions inside of CJS modules,
+// including via eval, all kinds of specifiers should work without issue.
+{
+  // Importing a built-in, both direct & via eval
+  assert.strictEqual(require('fs'), fs);
+  assert.strictEqual(eval('require("fs")'), fs);
+  assert.strictEqual(require('node:fs'), fs);
+  assert.strictEqual(eval('require("node:fs")'), fs);
+
+  assert.throws(
+    () => require('node:unknown'),
+    { code: 'ERR_UNKNOWN_BUILTIN_MODULE' },
+  );
+
+  assert.throws(
+    () => require('node:internal/test/binding'),
+    { code: 'ERR_UNKNOWN_BUILTIN_MODULE' },
+  );
+}
+
+// `node:`-prefixed `require(...)` calls bypass the require cache:
```

Comment:
this differs from other builtins, I love it, but want to be sure it is intentional.

Reply:
this actually seems like an issue - it means someone who replaces a builtin intentionally can’t replace one of these (altho presumably mutating a builtin would be visible in both). That could cause an issue where someone wants to use a package to instrument or lock down `fs` (and thus, policies are not an ergonomic option, unless I’m misunderstanding how policies work), but accidentally leaves a gaping security hole when an attacker requires `node:fs`.

Reply:
policies should intercept before any of the resolution helpers even get called https://github.com/nodejs/node/blob/36cc0ee993a07174f47ea598b6c2c75a66cb052c/lib/internal/modules/cjs/helpers.js#L52 , if any module redirection is in place (including complete shutdown) then it would never call out to `module.require` and get here without some kind of opt-in to the behavior (like `dependencies:true`).

Reply:
At the very least this needs documentation and probably a strong indication people using policies + instrumenting fs now need to update the policy.

Reply:
My point was that policies aren’t currently required to ensure this for CJS - a package (not the app itself) can do it. Forcing policies to be the mechanism means packages are no longer capable of abstracting this for users.

Reply:
@ljharb ah, yes that is true. I just wanted to be clear that policies can prevent this intentionally and should not be considered to have support for such a workflow.

Reply:
Totally understood. Given that policies are the superior mechanism for app developers to lock things down, what's the benefit of adding special, inconsistent, cache-bypassing behavior for `require` with a `node:` prefix?

Reply:
@ljharb i have no strong opinion on if we should diverge, but cache first behavior of CJS isn't shared by ESM and isn't robust against things like core destructuring the original impls. My like is just that it isn't a confusing situation like with `fs/`.

Reply:
Note that `import(…)` of built-in modules already bypasses the `require.cache`, e.g.:
```js
const fs = require('fs');

const fakeModule = {};
require.cache.fs = { exports: fakeModule };

assert.strictEqual((await import('fs')).default, fs);
```

Reply:
indeed, but that doesn’t mean it’s a good idea to make CJS inconsistent with itself.

Reply:
I don't think CJS is able to be claimed as consistent with itself since in general the natives don't normally populate the cache. I think it was just ad-hoc written together and the cache behavior is evolutionary not intentional or well understood (even by me).